rmgpy/tools/uncertainty.py

```diff
@@ -341,37 +326,65 @@ def loadModel(self, chemkinPath, dictionaryPath, transportPath=None):
                                                               dictionaryPath=dictionaryPath,
                                                               transportPath=transportPath)
 
+    def retrieveSaturatedSpeciesFromList(self,species):
+        """
+        Given a radical `species`, this function retrieves the saturated species objects from a list of species objects.
+        """
+        
+        molecule = species.molecule[0]
+        assert molecule.isRadical(), "Method only valid for radicals."
+        saturatedStruct = molecule.copy(deep=True)
+        saturatedStruct.saturate()
+        for otherSpecies in self.speciesList:
+            if otherSpecies.isIsomorphic(saturatedStruct):
+                return otherSpecies, False
+        else:
+            for key,value in self.database.thermo.libraries.iteritems():
+                for entry in value.entries.values():
+                    if entry.item.isIsomorphic(saturatedStruct):
+                        newSpc = Species().fromAdjacencyList(saturatedStruct.toAdjacencyList())
+                        newSpc.thermo = entry.data
+                        self.speciesList.append(newSpc)
+                        return newSpc, True
+                    
+            raise Exception('Could not retrieve saturated species form of {0} from the species list'.format(species))
 
     def extractSourcesFromModel(self):
         """
         Extract the source data from the model using its comments.
         Must be done after loading model and database to work.
         """
         self.speciesSourcesDict = {}
+        ignoreSpcs = []
         for species in self.speciesList:
-            source = self.database.thermo.extractSourceFromComments(species)
+            if not species in ignoreSpcs:
```

Comment:
Is this line necessary? I think the only way for a species to be added to `ignoreSpcs` is for it to have gone through this block once, and since we only loop over each specie in `self.speciesList` once this line should always be true.

Reply:
So ignoreSpcs are saturated species that are needed for getting thermo uncertainty correct, but aren't in the model, and they get appended to the end of speciesList so it can find them later.  However we do not want the sources for that molecule indexed in the same way species that are actually in the model are to avoid confusion later in processing.